internal/project/project.go

```diff
@@ -134,11 +135,25 @@ func findProjectsUnderPath(targetPath *paths.Path, projectTypeFilter projecttype
 	}
 
 	if recursive {
+		if symlinkDepth > 10 {
+			panic(fmt.Sprintf("symlink depth exceeded maximum while finding projects under %s", targetPath))
+		}
 		// targetPath was not a project, so search the subfolders.
 		directoryListing, _ := targetPath.ReadDir()
 		directoryListing.FilterDirs()
 		for _, potentialProjectDirectory := range directoryListing {
-			foundProjects = append(foundProjects, findProjectsUnderPath(potentialProjectDirectory, projectTypeFilter, recursive)...)
+			// It is possible for a combination of symlinks to parent paths to cause project discovery to get stuck in
+			// an endless loop of recursion. This is avoided by keeping count of the depth of symlinks and discontinuing
+			// recursion when it exceeds reason.
+			pathStat, err := os.Lstat(potentialProjectDirectory.String())
+			if err != nil {
+				panic(err)
+			}
+			if pathStat.Mode()&os.ModeSymlink != 0 {
+				symlinkDepth++
+			}
+
+			foundProjects = append(foundProjects, findProjectsUnderPath(potentialProjectDirectory, projectTypeFilter, recursive, symlinkDepth)...)
```

Comment:
There may be many symlinks in the same directory, so `symlinkDepth` may be increased multiple times in the same loop. It's very unusual but still...

```suggestion
			depthDelta := 0
			if pathStat.Mode()&os.ModeSymlink != 0 {
				depthDelta = 1
			}

			foundProjects = append(foundProjects, findProjectsUnderPath(potentialProjectDirectory, projectTypeFilter, recursive, symlinkDepth + depthDelta)...)
```

Reply:
Thanks so much @cmaglie!